django/test/testcases.py

```diff
@@ -1044,8 +1044,15 @@ def _fixture_setup(self):
     def _fixture_teardown(self):
         if not connections_support_transactions():
             return super(TestCase, self)._fixture_teardown()
+        for db_name in reversed(self._databases_names()):
+            if self._should_check_constraints(connections[db_name]):
+                connections[db_name].check_constraints()
         self._rollback_atomics(self.atomics)
 
+    def _should_check_constraints(self, connection):
+        return (connection.features.can_defer_constraint_checks and
```

Comment:
We try to avoid non 4-space indent. I'd do something like:
```
return (
    connection.features.can_defer_constraint_checks and
    not connection.needs_rollback and connection.is_usable()
)
```